### PR TITLE
Skip VTOL_TAKEOFF mission item when in fixed wing mode

### PR DESCRIPTION
**Pull request by RomanBapst**
## Describe problem solved by this pull request
When updating a mission while the vehicle is already flying in fixed wing mode the mission index is reset. If the mission contains a VTOL_TAKEOFF item then the vehicle will navigate to it. This does not make sense as the vehicle has already transitioned.

## Describe your solution
Skip VTOL_TAKEOFF item when already in fixed wing mode.

### CODE REVIEW

**Comment by RomanBapst** [comment]
Replaces #19287

File: src/modules/navigator/mission.cpp [hunk 1]
```diff
@@ -821,14 +821,10 @@ Mission::set_mission_items()
...
 
-					if (_navigator->get_vstatus()->vehicle_type == vehicle_status_s::VEHICLE_TYPE_ROTARY_WING) {
-						/* haven't transitioned yet, trigger vtol takeoff logic below */
-						_work_item_type = WORK_ITEM_TYPE_TAKEOFF;
 
-					} else {
-						/* already in fixed-wing, go to waypoint */
-						_mission_item.nav_cmd = NAV_CMD_WAYPOINT;
-					}
+					/* haven't transitioned yet, trigger vtol takeoff logic below */
```
Comment (sfuhrer) [inline src/modules/navigator/mission.cpp:825]: This code is still executed when already in FW mode, so the comment needs to be updated.

**Review by sfuhrer — CHANGES_REQUESTED** [review]
See comment.

But functionality-wise guarding against it in mission_item_to_position_setpoint()  seems fine to me.

File: src/modules/navigator/mission.cpp (hunk 1 above)
Comment (RomanBapst) [reply src/modules/navigator/mission.cpp]: @sfuhrer Done

**Review by sfuhrer — APPROVED** [review]

**Comment by hamishwillee** [comment]
@sfuhrer More generally, it would be good to clarify the expectation of a mission for VTOL. For example, a copter will take off if you start a mission that has no takeoff. It sounds like a VTOL requires one, but will skip it if you have already taken off (right).

Are there any other special rules you know about for VTOL? Mission items that work for everyone else but not VTOL. Mission items (other than transition and this takeoff that only work in VTOL). What happens if someone uses a normal takeoff item and not a vtol one)? 

Just hoping to better understand what people can and cannot do in missions.

**Comment by hamishwillee** [comment]
PS And does this require a change to MAVLInk definition of the command? I.e. is this something ever flight stack would have to do?

**Comment by sfuhrer** [comment]
>@sfuhrer More generally, it would be good to clarify the expectation of a mission for VTOL. For example, a copter will take off if you start a mission that has no takeoff. It sounds like a VTOL requires one, but will skip it if you have already taken off (right).

A VTOL behaves like a multicopter in a mission, unless you start with a MAV_CMD_NAV_VTOL_TAKEOFF or have a MAV_CMD_DO_VTOL_TRANSITION in your mission. 

> Mission items that work for everyone else but not VTOL

Not that I'm aware off. If the vehicle is in MC then it will execute the items as a MC, if in FW then as a FW. The weirdest may be Land and RTL -> MAV_CMD_NAV_LAND triggers a transition to MC (unless NAV_FORCE_VT is 0), and a MAV_CMD_NAV_RETURN_TO_LAUNCH (without a mission being present) will  cause the vehicle to loiter down around Home and then transition on RTL_DESCEND_ALT and descend in hover. 

>What happens if someone uses a normal takeoff item and not a vtol one

See above, will execute the mission as a MC.

>PS And does this require a change to MAVLInk definition of the command? I.e. is this something ever flight stack would have to do?

You mean skipping the item if already in air? Yes I think that's how it should be by definition.

**Pull request merged**